src/main/java/org/wso2/carbon/securevault/hashicorp/repository/HashiCorpSecretRepository.java

```diff
@@ -194,10 +237,11 @@ public SecretRepository getParent() {
     }
 
     /**
-     * Get the root token of the vault. Either by prompting the user via the console or by accessing the text file
-     * containing the root token.
+     * Get the secret id or root token of the vault. Either by prompting the user via the console or by accessing the
+     * text file
+     * containing the secret id.
      */
-    private void retrieveRootToken() throws HashiCorpVaultException {
+    private String retrieveAuthDetails(String credential) throws HashiCorpVaultException {
```

Comment:
Why do you want to pass the variable into the method, return it from the method and assign it again from the place that is calling the method?

Reply:
We are using this retriveAuthDetials method for retrieve both root token and secrete id. Therefor we are passing the variable to the method and update it during retrieval

Reply:
I'm not quite sure what you are trying to explain here.

Reply:
Corrected this in 809f4a8dd11da47c8e69b277377c547f2d3549d7